Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/sysupdate/ui/SystemUpdateProgressBarDialogFragment.java

```diff
@@ -28,27 +32,25 @@ public Dialog onCreateDialog(@Nullable Bundle savedInstanceState)
     SystemUpdateViewModel viewModel =
             new ViewModelProvider(requireActivity()).get(SystemUpdateViewModel.class);
 
-    ProgressDialog progressDialog = new ProgressDialog(requireContext());
-    progressDialog.setTitle(getString(R.string.updating));
+    ProgressBar progressBar =
+            new ProgressBar(requireContext(), null, android.R.attr.progressBarStyleHorizontal);
+    final int dpHorizontal = 24;
+    final int dpVertical = 16;
+    int pxHorizontal = ThemeHelper.dpToPx(dpHorizontal, getResources());
+    int pxVertical = ThemeHelper.dpToPx(dpVertical, getResources());
+    progressBar.setPadding(pxHorizontal, pxVertical, pxHorizontal, pxVertical);
+
     // We need to set the message to something here, otherwise the text will not appear when we set it later.
-    progressDialog.setMessage("");
-    progressDialog.setButton(Dialog.BUTTON_NEGATIVE, getString(R.string.cancel), (dialog, i) ->
-    {
-    });
-    progressDialog.setOnShowListener((dialogInterface) ->
-    {
-      // By default, the ProgressDialog will immediately dismiss itself upon a button being pressed.
-      // Setting the OnClickListener again after the dialog is shown overrides this behavior.
```

Comment:
Is this not needed anymore? Or why was it removed?

Reply:
Looks like it is still needed. This is one way you can prevent an AlertDialog from dismissing when pressing a button and replace it with custom behavior. I have reimplemented this in the latest commit. (It's slightly different, but it does the same thing)